### ISSUE TITLE
rootfs tarball

### ISSUE THREAD

**Issue by NeQuissimus** [NixOS/nixpkgs#19522]
I would like to build a NixOS `Dockerfile` from scratch.
For this, it is easiest to have a `rootfs.tar.xz`.
Running something like `NIXOS_CONFIG=/home/nequi/dev/DevSetup/docker/configuration.nix nix-build -A config.system.build.tarball --argstr system x86_64-linux` will give me a tarball, which I assume to be the rootfs under a different name.

This tarball does not quite work for me but that is not what this is about.

Is there a place on Hydra that keeps the tarballs used to create the ISOs for official releases?
I tried looking through some evaluations but could not find anything.

**Comment by dezgeg** [comment]
ISO creation doesn't use an intermediate tarball. It's built from the system closure (`config.system.build.toplevel`) directly.

**Comment by NeQuissimus** [comment]
Oh, I see.

So the output of nixos/release.nix -> makeSystemTarball is not actually generated?

**Comment by dezgeg** [comment]
Oh, indeed we actually generate those: http://hydra.nixos.org/build/42024611. But it is not used by the ISO image build, it's something separate (and quite likely not covered by any NixOS tests).

**Comment by NeQuissimus** [comment]
Ah, this is what I was looking for :D
Except, the file seems to get deleted  :(

**Comment by florianjacob** [comment]
Yeah, this tarball seems to vanish instantly after generation. ☹️ I used the „reproduce this build“ link and compiled it myself, problem is the resulting tarball seems to only contain /nix/store and some mountpoints but no something like /etc/os-release (or any /etc at all) so that I could execute it with e.g. systemd-nspawn.

**Comment by globin** [comment]
nixos/hydra#321

**Comment by mmahut** [comment]
Are there any updates to this issue, please?

**Comment by cedws** [comment]
I'm still interested in this. My use case for this is that I want to bootstrap a minimal image which is 'ready to go'. Kind of like how you can with debootstrap. I realise I could use something like Packer to run the installation ISO but I'm not too keen on this.

**Comment by theoparis** [comment]
I am also interested in this. I would like to create a root filesystem tarball of a NixOS system flake that can be extracted on a android device and booted via a custom, fastboot-compatible boot.img file.
